tests/lib/rules/prefer-expect-query-by.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/prefer-expect-query-by');
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2015, sourceType: 'module' },
+});
+
+ruleTester.run('prefer-expect-query-by', rule, {
+  valid: [
+    { code: "expect(queryByText('Hello')).not.toBeInTheDocument()" },
+    { code: "expect(rendered.queryByText('Hello')).not.toBeInTheDocument()" },
+    { code: "expect(queryAllByText('Hello')).not.toBeInTheDocument()" },
+    {
+      code: "expect(rendered.queryAllByText('Hello')).not.toBeInTheDocument()",
+    },
+  ],
+  invalid: [
+    {
+      code: "expect(getByText('Hello')).not.toBeInTheDocument()",
+      errors: [{ messageId: 'expectQueryBy' }],
+    },
+    {
+      code: "expect(rendered.getByText('Hello')).not.toBeInTheDocument()",
+      errors: [{ messageId: 'expectQueryBy' }],
+    },
+    {
+      code: "expect(getAllByText('Hello')).not.toBeInTheDocument()",
+      errors: [{ messageId: 'expectQueryBy' }],
+    },
+    {
+      code: "expect(rendered.getAllByText('Hello')).not.toBeInTheDocument()",
+      errors: [{ messageId: 'expectQueryBy' }],
+    },
+  ],
+});
```

Comment:
What if we have:
```
const message = getByText('Hello')
expect(message).toBeInTheDocument()
```

Is that case reported? If not, @thomlom can help you with that as he implemented that check for other previous rules.

Reply:
From my understanding, doing it like this would throw and exit the test before the expect is called, in case the element is not found.
I think that usually the problem is when you do that within the `expect`.

Reply:
This is a really good point, it's fine to remain like this then.

Reply:
@thomlom I would leave it without variable declarator as @emmenko did a good point about it so I think it should be only checked when querying directly within expect statement.